src/command.rs

```diff
@@ -188,7 +200,11 @@ pub async fn publish(
 
     let artifactory = Artifactory::new(registry, &credentials)?;
 
-    let package = PackageStore::release().await?;
+    let manifest = Manifest::read().await?;
+    let package = PackageStore::current()
+        .into_diagnostic()?
```

Comment:
Same here

Reply:
Same reason: `current()` only checks some stuff that may return `std::io::Error`. We always try to return the most specific error type if we can.

`miette::Result` does not work like `eyre::Result` or `anyhow::Result`. You cannot just cast any error into a `miette::Error` with `?`, you have to explicitly convert it using `into_diagnostic()`.

We don't want to change the return error of `PackageStore::current()`, because then we would make the errors less useful and we would make testing harder than it needs to be. In my understanding, `miette::Result` is mostly useful for pretty-printing errors.

So we will have to deal with these `into_diagnostic` calls in certain places, as far as I can tell.